web/src/main/java/org/openmrs/web/controller/encounter/EncounterDisplayController.java

```diff
@@ -387,6 +387,42 @@ public int compareTo(FieldHolder other) {
 		}
 		
 		/**
+		 * Indicates whether some other object is "equal to" this one.
+		 *
+		 * @see java.lang.Object#equals(java.lang.Object)
+		 */
+		public boolean equals(FieldHolder other) {
+			if ((other == null) || (getClass() != other.getClass())) {
+				return false;
+			} // end if
+			else {
+				if (formField == null) {
+					if (other.getFormField() != null) {
+						return false;
+					}
+				} else if (!formField.equals(other.getFormField())) {
+					return false;
+				}
+				if (obs == null) {
+					if (other.getObs() != null) {
+						return false;
+					}
+				} else if (!obs.equals(other.getObs())) {
+					return false;
+				}
+				if (groupMemberConcepts == null) {
+					if (other.getGroupMemberConcepts() != null) {
+						return false;
+					}
+				} else if (!groupMemberConcepts.equals(other.getGroupMemberConcepts())) {
+					return false;
+				}
+			} // end else
+			
+			return true;
+		} //
```

Comment:
Whenever you implement equals you must implement hashCode. Also read JavaDoc for compareTo. If equals returns true then compareTo must return 0. If equals returns false then compareTo must return != 0. You need to correct equals to behave accordingly. In other words it's enough to implement equals as return compareTo() == 0.